lib/pbench/cli/server/report.py

```diff
@@ -0,0 +1,403 @@
+            if not size:
+                detailer.write(f"{tarball.name} has no unpacked size")
+                lacks_size += 1
+            elif not isinstance(size, int):
+                detailer.write(
+                    f"{tarball.name} has non-integer unpacked size "
+                    f"{size!r} ({type(size)})"
+                )
+                bad_size += 1
+            else:
+                if not smallest_cache or size < smallest_cache:
+                    smallest_cache = size
+                    smallest_cache_name = tarball.name
+                if not biggest_cache or size > biggest_cache:
+                    biggest_cache = size
+                    biggest_cache_name = tarball.name
+                cached_size += size
+    oldest = datetime.datetime.fromtimestamp(oldest_cache, datetime.timezone.utc)
+    newest = datetime.datetime.fromtimestamp(newest_cache, datetime.timezone.utc)
+    click.echo("Cache report:")
+    click.echo(
+        f"  {cached_count} datasets are cached, consuming "
+        f"{humanize.naturalsize(cached_size)}"
+    )
+    click.echo(
+        f"  {lacks_size} datasets have never been unpacked, "
+        f"{last_ref_errors} are missing reference timestamps, "
+        f"{bad_size} have bad size metadata"
+    )
+    click.echo(
+        f"  The smallest cache, {smallest_cache_name}, is "
+        f"{humanize.naturalsize(smallest_cache)}"
+    )
+    click.echo(
+        f"  The biggest cache, {biggest_cache_name}, is "
+        f"{humanize.naturalsize(biggest_cache)}"
+    )
+    click.echo(
+        f"  The least recently used cache, {oldest_cache_name}, was "
+        f"referenced {humanize.naturaldate(oldest)}"
+    )
+    click.echo(
+        f"  The most recently used cache, {newest_cache_name}, was "
+        f"referenced {humanize.naturaldate(newest)}"
+    )
+
+
+def report_sql():
+    """Report the SQL table storage statistics"""
+    click.echo("SQL storage report:")
+    click.echo("  Table                Rows       Storage")
+    click.echo("  -------------------- ---------- ----------")
+    for t in inspect(Database.db_session.get_bind()).get_table_names():
+        rows = list(
+            Database.db_session.execute(statement=text(f"SELECT COUNT(*) FROM {t}"))
+        )[0][0]
+        size = list(
+            Database.db_session.execute(
+                statement=text("SELECT pg_total_relation_size(:table)"),
+                params={"table": t},
+            )
+        )[0][0]
+        click.echo(f"  {t:20} {rows:>10} {humanize.naturalsize(size):>10}")
+
+    if not detailer:
+        return
+
+    query = select(IndexMap.root, IndexMap.index)
+    idxes = Database.db_session.execute(query).all()
+    record_count = 0
+    roots = set()
+    indices = set()
+    root_size = 0
+    index_size = 0
+    for idx in idxes:
+        record_count += 1
+        roots.add(idx[0])
+        indices.add(idx[1])
+        root_size += len(idx[0])
+        index_size += len(idx[1])
```

Comment:
It would be nicer if you unpacked `idx`:
```python
    for root, index in idxes:
        record_count += 1
        roots.add(root)
        indices.add(index)
        root_size += len(root)
        index_size += len(index)
```